linode/manager
# upcoming: [M3-7949] - Gecko GA Region Select

## Description 📝
Add tabbed Core/Distributed Region select for the Linode Create flow for Gecko GA. Since distributed regions are in a separate tab, there is no need to display the distributed icon. The region label also now displays Country, City with the slug id right aligned

_Note: Regions in other areas of the app may still display as City, State. This will be addressed separately in M3-8209_

## Changes  🔄
- Added tabbed Core/Distributed Region select in the Linode Create flow
- Hide distributed icon for Gecko GA
- Display region label as Country, City instead of City, State
- Region slug id is now right aligned

## Preview 📷
| Before  | After   |
| ------- | ------- |
| <video src="https://github.com/linode/manager/assets/115299789/7aed0b46-d9b8-4c46-8408-96fa80ea320c" /> | <video src="https://github.com/linode/manager/assets/115299789/2c92e862-8c18-4a22-913f-a017ac1fa256" /> |

## How to test 🧪

### Prerequisites
- Ensure your account has the `new-dc-testing`, `edge_testing` and `edge_compute` customer tags

### Verification steps
- Go to the Linode Create flow, the Region section should now have two tabs, `Core` and `Distributed`
- The region select options for both Core and Distributed should display as Country, City with the slug id right aligned
- The region select options for the Core tab should only show regions with a `site_type` of `core`
  - Selecting a core region should display Linode plans as normal
- The region select options for the Distributed tab should only show regions with a `site_type` of `distributed`
  - Additionally, users can filter distributed regions by geographical area. The geographical area should default to `All`
  - Selecting a geographical area should filter the region select for regions in the geographical area. If `All` is selected, display all distributed regions
  - Selecting a distributed region should show the distributed plan table with warning notice about pricing
- The tabbed regions should not display if the customer does not have the customer tags
- Ensure that Gecko Beta still functions as expected. We want to be able to toggle between Beta and GA seamlessly.
  - Comment out the hard coded gecko2 test values in `useFlags.ts` and in the devtools click on `Reset to LD default flags`

```
yarn test RegionSelect.utils
```

## As an Author I have considered 🤔

*Check all that apply*
